pavelib/assets.py

```diff
@@ -384,14 +384,36 @@ def coffeescript_files():
 
 @task
 @no_help
-def compile_coffeescript(*files):
+def compile_coffeescript(*coffee_files):
     """
     Compile CoffeeScript to JavaScript.
+
+    Arguments:
+        coffee_files: (optional) packed list of files (passed as individual args) to be converted from .coffee to .js
+    Example:
+        compile_coffeescript("file1.js", "file2.js", "file3.js")
+
+    If no args are passed, the Coffeescript binary will instead be passed a bash substitution that,
+    when expanded, lists all Coffeescript files in the project. All Coffeescript in the project will be compiled.
     """
-    if not files:
-        files = ["`{}`".format(coffeescript_files())]
+
+    coffee_find_cmd = coffeescript_files()
+    js_find_cmd = r"$({} | sed 's/\.coffee/\.js/g')"
+
+    if not coffee_files:
+        coffee_files = ["$({})".format(coffee_find_cmd)]
+        js_files = js_find_cmd.format(coffee_find_cmd)
+    else:
+        js_files = js_find_cmd.format(*coffee_files)
```

Comment:
This won't work if there's more than one file in `coffee_files`. The `js_find_command` only has a single slot to format into.